### PR TITLE
Addition of <segments> concept for multi-screen devices 

### PR DESCRIPTION
**Pull request by ststimac**
+@dlibby-
-@ststimac

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -256,6 +258,23 @@ <h3>The <dfn><code>VisualViewport</code></dfn> interface</h3>
                 Although it is referred to as the pinch-zoom scaling factor, it can be affected through means other than pinch-zooming. e.g. When the UA centers and zooms in a focused edit box.
               </div>
             </dd>
+            <dt><dfn>segments</dfn></dt>
+            <dd>
+              <p>
+                Returns an array of DOMRects that represent the dimensions of each existing viewport segment. 
+              </p>
+              <p>
+                A viewport segment represents the regions (and their dimensions) of the visual viewport that reside on separate displays or logical display regions.
```
Comment (bokand) [inline index.html]: Nit:

```
A viewport segment represents the regions
```

"A viewport segment" implies a single element of the array so I'd expect that to represent a single "region". I'd rephrase this to something like (unless I'm misunderstanding, please correct me if I'm wrong):

```
Viewport segments represent the regions...
```
or
```
A viewport segment represents a region of the visual viewport that resides on a separate display...
```

---

```
(and their dimensions)
```
Can this be removed? Doesn't a "region" imply dimensions?

**Review by bokand — COMMENTED** [review]
This looks good to me but I'm not up to date on this proposal so could someone familiar with it review for correctness? Maybe @dlibby-?

Is there some discussion in a working group or similar we could link to?

Also, is there any way we can mark this (as Daniel mentioned) "more experimental" than the current `visualViewport`?

**Comment by ststimac** [comment]
@bokand I'm making the updates to this PR now. This is the link for the discussion we have for these updates: https://www.w3.org/2021/02/24-webscreens-irc

This is my first time editing a spec/draft spec. Where's the appropriate location to drop this link? The other links section at the top?

File: index.html [hunk 2]
```diff
@@ -29,6 +29,9 @@
                 }, {
                     value: 'Feedback / discussion in the WICG',
                     href: 'https://discourse.wicg.io/t/explicit-api-for-visual-viewport/1399'
+                }, {
+                    value: 'Viewport segments discussion',
+                    href: 'https://www.w3.org/2021/02/24-webscreens-irc'
```
Comment (karlcow) [inline index.html]: nit: Maybe this link is a bit easier on the eyes. This is the html version of the minutes.
https://www.w3.org/2021/02/24-webscreens-minutes

(sorry for the drive-by comment, and you can ignore it, if you prefer the text version)

File: index.html [hunk 3]
```diff
@@ -256,6 +261,26 @@ <h3>The <dfn><code>VisualViewport</code></dfn> interface</h3>
                 Although it is referred to as the pinch-zoom scaling factor, it can be affected through means other than pinch-zooming. e.g. When the UA centers and zooms in a focused edit box.
               </div>
             </dd>
+            <dt><dfn>segments</dfn></dt>
+            <dd>
```
Comment (dlibby-) [inline index.html:265]: Per discussion, can we add a `Note:` here about this API being experimental or that it is an addition to the 'base APIs' that are currently shipping?

File: index.html [hunk 4]
```diff
@@ -256,6 +261,26 @@ <h3>The <dfn><code>VisualViewport</code></dfn> interface</h3>
...
+              <p>
+                <code>segements</code> is null when there is only a single viewport segment. Returns null when called from within an iframe context.
+              </p>
+            </dd>
+            <div class="note">
+              Based on the data returned for each viewport segment, developers will be able to infer the number of hinges available as well as the hinge orientation.
+              The browser window can be moved/resized such that the number of segments and/or their dimensions change. In these cases, resize events will fire at which point authors should re-query this property, as the returned FrozenArray is just a snapshot of the current state. 
+            </div>
+            <div class="note">
+              The viewport segments code is currently in development and considered experimental.
```
Comment (dlibby-) [inline index.html]: nit: should read the "segments property" or something similar instead of referencing 'code'

**Review by dlibby- — APPROVED** [review]
overall LGTM, just one nit in the `note:`

**Comment by ststimac** [comment]
Updates made per @dlibby-'s review and made the link update to the webscreen minutes, thanks @karlcow!
